src/Interpreters/getColumnFromBlock.cpp

```diff
@@ -52,7 +52,7 @@ ColumnPtr tryGetSubcolumnFromBlock(const Block & block, const DataTypePtr & requ
         return elem_column;
     }
 
-    auto elem_column = elem->type->tryGetSubcolumn(subcolumn_name, elem->column);
+    auto elem_column = elem->type->tryGetSubcolumn(subcolumn_name, elem->column->decompress());
```

Comment:
We do the same in `tryGetColumnFromBlock()` above, so I didn't think much about it